### PR TITLE
Issue/5677 Add search to blog selector

### LINKED ISSUE #5677: Editor: Blog selector modal should be searchable
The blog selector modal / popover in the editor should be searchable, like `BlogListViewController`. As well as making it harder to find a site in a long list, there's currently no way to select hidden sites from the selector. The search functionality should behave the same as the blog list: when the search is activated, hidden sites become visible for the duration of the search.

### PR DESCRIPTION
**Pull request by frosty**
Fixes #5677. I've added search to the blog selector modal in the editor, and updated the style of the tableview to match the recently-updated My Sites list.

To test:

* Open the editor, and open the blog selector.
* Please ensure you test the top and bottom tableview scroll indicator and content insets when showing / hiding the keyboard and activating / cancelling search on both iPhone and iPad, in different 
orientations and multitasking configurations.

Needs review: @kurzee

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m [hunk 1]
```diff
@@ -378,4 +498,49 @@ - (NSArray *)sortDescriptorsWithAccountKeyPath
...
+    if (![self.resultsController performFetch:&error]) {
+        DDLogError(@"Couldn't fetch sites: %@", [error localizedDescription]);
+    }
+
+    [self.tableView reloadData];
+}
+
+// Improves the appearance of the arrow when displayed in a popover, by matching
+// its color to the topmost view in the scrollview
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m]: This is super neat, but it does break a bit within the rect of the search field's white bg color.

screen shot 2016-07-18 at 1 29 13 pm

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 2]
```diff
@@ -739,14 +739,33 @@ - (void)showBlogSelector
                                                                                        dismissHandler:dismissHandler];
     vc.title = NSLocalizedString(@"Select Site", @"");
     vc.displaysPrimaryBlogOnTop = YES;
-    vc.displaysCancelButton = [self hasHorizontallyCompactView];
-    UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:vc];
+
+    if ([UIDevice isPad] && ![self hasHorizontallyCompactView]) {
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: Let's go with the helper `[WPDeviceIdentification isiPad]` for this one.

**Comment by kurzee** [comment]
@frosty this is great! Only two nitpicky comments, :shipit:!

File: WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m (hunk 1 above)
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m]: As discussed, I'm going to push this as-is for now. I'd rather not hard code assumptions about the position of the text field in the search bar (there's no way to access it directly), and I feel that at least doing _something_ is better than the default. :)

**Pull request merged**